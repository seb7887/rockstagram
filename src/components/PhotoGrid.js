import React from 'react';
import styled from 'styled-components';

import Photo from './Photo';

const Grid = styled.div`
  display: flex;
  flex-wrap: wrap;
  max-width: 1200px;
  margin: 0 auto;
`;

class PhotoGrid extends React.Component {
  render() {
    const { posts } = this.props;
    return (
      <Grid>
        {posts.map((post, i) => <Photo {...this.props} key={i} i={i} post={post} />)}
      </Grid>
    );
  }
}

export default PhotoGrid;
