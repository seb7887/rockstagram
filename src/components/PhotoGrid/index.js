import React from 'react';
import { Grid } from './style';

import Photo from '../Photo';

class PhotoGrid extends React.Component {
  render() {
    const { posts } = this.props;
    return (
      <Grid>
        {posts.map((post, i) => (
          <Photo {...this.props} key={i} i={i} post={post} />
        ))}
      </Grid>
    );
  }
}

export default PhotoGrid;
