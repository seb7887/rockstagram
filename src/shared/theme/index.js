const theme = {
  colors: {
    black: '#0b0b0b',
    offwhite: '#ffffff',
    white: 'white',
    lightgrey: '#fafafa',
    red: 'red',
  },
  fonts: {
    // eslint-disable-next-line quotes
    header: `billabong, 'billabongregular'`,
  },
};

export default theme;
