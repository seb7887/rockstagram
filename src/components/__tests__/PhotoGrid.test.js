/* eslint-env jest */
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from 'react-testing-library';

import PhotoGrid from '../PhotoGrid';
import theme from '../../shared/theme';

const comments = {
  test: [
    {
      text: 'hey!',
      user: 'kingmob',
    },
  ],
};

const post = {
  caption: 'This is a test',
  code: 'test',
  display_src: 'test.jpg',
  id: '7',
  likes: 7,
};

const posts = [post];

const incrementLikes = jest.fn();

const mockedProps = {
  posts,
  incrementLikes,
  comments,
};

describe('<PhotoGrid/>', () => {
  it('renders and matches snapshot', () => {
    const { asFragment } = render(
      <Router>
        <ThemeProvider theme={theme}>
          <PhotoGrid {...mockedProps} />
        </ThemeProvider>
      </Router>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
