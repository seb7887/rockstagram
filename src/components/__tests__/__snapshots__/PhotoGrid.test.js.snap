// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoGrid/> renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="style__Grid-sc-1if1ioi-0 RDiNo"
  >
    <figure
      class="style__Figure-cjo9le-0 kWuZVR"
    >
      <div
        class="grid-photo-wrap"
      >
        <a
          href="/view/test"
        >
          <img
            alt="This is a test"
            class="grid-photo"
            src="test.jpg"
          />
        </a>
        <div
          class="LikeAnimation__Animation-sc-1q1izn2-0 fqDDFP"
        >
          <div>
            <span
              class="like LikeAnimation__Likes-sc-1q1izn2-1 kCBfRn"
            >
              7
            </span>
          </div>
        </div>
      </div>
      <figcaption
        data-testid="caption"
      >
        <p>
          This is a test
        </p>
        <div
          class="control-buttons"
        >
          <button
            class="likes"
          >
            <i
              class="fa fa-heart"
              data-testid="likes"
            />
             7
          </button>
          <a
            class="button"
            href="/rockstagram/view/test"
          >
            <span
              class="comment-count"
              data-testid="count"
            >
              <i
                class="fa fa-comment"
              />
               1
            </span>
          </a>
        </div>
      </figcaption>
    </figure>
  </div>
</DocumentFragment>
`;
